watertap/core/membrane_channel_base.py

```diff
@@ -519,16 +562,36 @@ def eq_K(b, t, x, j):
                 == b.properties[t, x].diffus_phase_comp["Liq", j] * b.N_Sh_comp[t, x, j]
             )
 
-        @self.Constraint(
-            self.flowsheet().config.time,
-            self.length_domain,
-            self.config.property_package.solute_set,
-            doc="Sherwood number",
-        )
-        def eq_N_Sh_comp(b, t, x, j):
-            return (
-                b.N_Sh_comp[t, x, j]
-                == 0.46 * (b.N_Re[t, x] * b.N_Sc_comp[t, x, j]) ** 0.36
+        if sherwood_number_eq == SherwoodNumberEq.length_dependent:
+
+            @self.Constraint(
+                self.flowsheet().config.time,
+                self.length_domain,
+                self.config.property_package.solute_set,
+                doc="Sherwood number",
+            )
+            def eq_N_Sh_comp_length_dependent(b, t, x, j):
+                return b.N_Sh_comp[t, x, j] == 2.401 * (
+                    (b.N_Re[t, x] * b.N_Sc_comp[t, x, j]) ** 0.297
+                ) * ((b.length / b.dh) ** -0.279)
```

Comment:
I still have the same concern as before: https://github.com/watertap-org/watertap/pull/872#discussion_r1082782345.

Here is the equation in the reference (https://www.sciencedirect.com/science/article/pii/S1383586622016768)
<img width="449" alt="Screenshot 2023-02-01 at 8 43 19 AM" src="https://user-images.githubusercontent.com/30801372/216090702-bbb39476-ec4e-4fce-bb4e-4b314bc98ced.png">

In the Nomenclature section the channel length is represented by _L_. So I think _x_ here is the "location along the membrane", e.g., `x`. Unfortunately this isn't stated in the nomenclature, but is at the bottom left on page 4:
<img width="448" alt="Screenshot 2023-02-01 at 8 50 10 AM" src="https://user-images.githubusercontent.com/30801372/216093321-09553aad-4ec7-4e0b-af3f-0ab32e606680.png">

Further, if my supposition is correct and we need to change this equation, then we need to carefully consider the "backwards" case, e.g., when `x` starts at `1` and goes to `0`.

Reply:
For the backwards case, I'm confused about this because Sherwood number isn't that only length-dependent variable. Maybe it's only a concern here because x is actually in the constraint, unlike (most of?) our other length-dependent variables. 

In any case, @MarcusHolly instead of `b.length` in the numerator, it should probably be something like `x * b.length` since x will be some value in the normalized length domain (0 to 1) and multiplying by length would yield the actual location. We can discuss today during our group call.

Reply:
> Maybe it's only a concern here because x is actually in the constraint, unlike (most of?) our other length-dependent variables. 

I think this is correct, but it's probably worth double checking to make sure it's the only one that would explicitly have `x` in the constraint.